From a0bc550645eb567f238c693e766efde2f6efd98d Mon Sep 17 00:00:00 2001
From: Wang ShaoBo <bobo.shaobowang@huawei.com>
Date: Fri, 31 Dec 2021 15:59:44 +0800
Subject: [PATCH] kabi: reserve space for kobject related structures
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: a0bc550645eb567f238c693e766efde2f6efd98d


hulk inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I4KFY7?from=project-issue
CVE: NA

-------------------------------

Reserve space for the structures in kobject subsystem.

Signed-off-by: Wang ShaoBo <bobo.shaobowang@huawei.com>
Reviewed-by: Cheng Jian <cj.chengjian@huawei.com>
Reviewed-by: Cheng Jian <cj.chengjian@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 include/linux/kobject.h | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/include/linux/kobject.h b/include/linux/kobject.h
index ea30529fba08..f3c9ecb71642 100644
--- a/include/linux/kobject.h
+++ b/include/linux/kobject.h
@@ -15,6 +15,7 @@
 #ifndef _KOBJECT_H_
 #define _KOBJECT_H_
 
+#include <linux/kabi.h>
 #include <linux/types.h>
 #include <linux/list.h>
 #include <linux/sysfs.h>
@@ -77,6 +78,10 @@ struct kobject {
 	unsigned int state_add_uevent_sent:1;
 	unsigned int state_remove_uevent_sent:1;
 	unsigned int uevent_suppress:1;
+	KABI_RESERVE(1)
+	KABI_RESERVE(2)
+	KABI_RESERVE(3)
+	KABI_RESERVE(4)
 };
 
 extern __printf(2, 3)
@@ -143,6 +148,10 @@ struct kobj_type {
 	const struct kobj_ns_type_operations *(*child_ns_type)(struct kobject *kobj);
 	const void *(*namespace)(struct kobject *kobj);
 	void (*get_ownership)(struct kobject *kobj, kuid_t *uid, kgid_t *gid);
+	KABI_RESERVE(1)
+	KABI_RESERVE(2)
+	KABI_RESERVE(3)
+	KABI_RESERVE(4)
 };
 
 struct kobj_uevent_env {
@@ -194,6 +203,10 @@ struct kset {
 	spinlock_t list_lock;
 	struct kobject kobj;
 	const struct kset_uevent_ops *uevent_ops;
+	KABI_RESERVE(1)
+	KABI_RESERVE(2)
+	KABI_RESERVE(3)
+	KABI_RESERVE(4)
 } __randomize_layout;
 
 extern void kset_init(struct kset *kset);
-- 
2.26.2

